### DISCUSSION TITLE
Implementation of PESegmentReference::GetSegment()

### DISCUSSION THREAD

**Discussion by dashne** [Washi1337/AsmResolver#600] [Ideas]
Is there any chance in future, implementation of this line : (`public ISegment? GetSegment() => throw new InvalidOperationException();`) in [PESegmentReference] , specially for methodbody, would be very helpful for binary patch.

thanks.

**Comment by Washi1337** [comment]
Very likely no.

This has been a conscious design decision in the way the references system works.  References into the PE itself are often unbound, and given an arbitrary reference (usually RVA), it is impossible to resolve it to a concrete segment without the context in which this reference is used. Unless we can come up with a design that allows for this context to be packaged within with the reference, I don't see this changing soon.

Usually if you must rely on the concrete segment, you can and probably should get it in some other way.
